src/core/atomic.d

```diff
@@ -1273,6 +1182,146 @@ else version( AsmX86_64 )
     }
 }
 
+/* ***
+ * Functions built in to the compiler, along with a default implementation.
+ */
+
+bool cmpxchg(shared(ubyte)* here, ubyte ifThis, ubyte writeThis) pure nothrow @nogc @trusted
```

Comment:
`cas` adds another layer. Again, a systems programming language should expose the primitives, not hide them.